name_mapping.go

```diff
@@ -130,6 +130,122 @@ func visitMappedFields[S, T any](fields []MappedField, visitor NameMappingVisito
 	return visitor.Fields(fields, results)
 }
 
+func UpdateNameMapping(nameMapping NameMapping, updates map[int]NestedField, adds map[int][]NestedField) (NameMapping, error) {
+	result, err := VisitNameMapping(nameMapping, &updateNameMappingVisitor{updates: updates, adds: adds})
```

Comment:
Add a docstsring comment here to explain the benefits of using this vs just `CreateNameMapping` etc.